src/wmtk/EdgeCollapsingConn.cpp

```diff
@@ -4,18 +4,140 @@
 #include <wmtk/TetMesh.h>
 
 #include <wmtk/utils/TupleUtils.hpp>
+#include <wmtk/utils/VectorUtils.h>
+
+#include <spdlog/spdlog.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <limits>
+#include <type_traits>
 
 bool wmtk::TetMesh::collapse_edge(const Tuple& loc0, std::vector<Tuple>& new_edges)
 {
     if (!collapse_before(loc0)) return false;
 
+    auto link_condition = [&VC = this->m_vertex_connectivity,
+                           &TC = this->m_tet_connectivity,
+                           this](auto v0, auto v1) -> bool {
+        auto intersects = [](const auto& vec, const auto& val) {
+            for (auto& v : vec)
+                for (auto& a : val)
+                    if (v == a) return true;
+            return false;
+        };
+
+        auto link = [&intersects,
```

Comment:
this too

Reply:
this results in a crazy long function signature.